src/ore/src/vec.rs

```diff
@@ -167,22 +171,27 @@ impl<T> VecExt<T> for Vec<T> {
             pred: filter,
         }
     }
+
+    // implementation is from Vec::is_sorted_by, but with `windows` instead of
+    // the unstable `array_windows`
+    fn is_sorted_by<F>(&self, mut compare: F) -> bool
+    where
+        F: FnMut(&T, &T) -> bool,
+    {
+        self.windows(2).all(|win| compare(&win[0], &win[1]))
+    }
```

Comment:
It would be good to have a few test cases for this.

Reply:
Done